flink-clients/src/main/java/org/apache/flink/client/deployment/application/EmbeddedJobClient.java

```diff
@@ -126,7 +126,8 @@ public JobID getJobID() {
 					try {
 						return jobResult.toJobExecutionResult(classLoader);
 					} catch (Throwable t) {
-						throw new CompletionException(new Exception("Job " + jobId + " failed", t));
+						throw new CompletionException(
+								ApplicationFailureException.fromJobResult(jobResult, classLoader));
```

Comment:
Is this the actual change which fixes the problem?

Reply:
Yes.